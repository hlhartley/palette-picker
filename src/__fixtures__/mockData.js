export const mockProjects = [
    {
        id: 1,
        name: "Warm colors project",
        created_at: "2019-03-26T05:17:02.689Z",
        updated_at: "2019-03-26T05:17:02.689Z"
    },
    {
        id: 2,
        name: "Cool colors project",
        created_at: "2019-03-26T05:17:02.702Z",
        updated_at: "2019-03-26T05:17:02.702Z"
    },
    {
        id: 3,
        name: "Neutral colors project",
        created_at: "2019-03-26T05:17:02.727Z",
        updated_at: "2019-03-26T05:17:02.727Z"
    }
]

export const mockPalettes = [
    {
        id: 1,
        project_id: 1,
        name: "Summer colors",
        color1: "FFFF82",
        color2: "DDDCF7",
        color3: "B5D99C",
        color4: "FFAF66",
        color5: "E65F5C",
        created_at: "2019-03-26T05:17:02.735Z",
        updated_at: "2019-03-26T05:17:02.735Z"
    },
    {
        id: 2,
        project_id: 1,
        name: "Spring colors",
        color1: "F25469",
        color2: "BD2D87",
        color3: "D664BE",
        color4: "DF99F0",
        color5: "B191FF",
        created_at: "2019-03-26T05:17:02.736Z",
        updated_at: "2019-03-26T05:17:02.736Z"
    },
    {
        id: 3,
        project_id: 2,
        name: "Fall colors",
        color1: "FCD0A1",
        color2: "B1B695",
        color3: "A690A4",
        color4: "EFA958",
        color5: "AFD2E9",
        created_at: "2019-03-26T05:17:02.751Z",
        updated_at: "2019-03-26T05:17:02.751Z"
    },
    {
        id: 4,
        project_id: 3,
        name: "Winter colors",
        color1: "594157",
        color2: "726DA8",
        color3: "7D8CC4",
        color4: "A0D2DB",
        color5: "BEE7E8",
        created_at: "2019-03-26T05:17:02.757Z",
        updated_at: "2019-03-26T05:17:02.757Z"
    }
]

export const mockHexcodes = [
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    }
]