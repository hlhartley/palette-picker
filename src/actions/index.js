export const setError = (error) => ({
  type: 'SET_ERROR', 
  error
});

export const setLoading = (isLoading) => ({
  type: 'SET_LOADING', 
  isLoading
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS', 
  projects
});

export const setPalettes = (palettes) => ({
  type: 'SET_PALETTES', 
  palettes
});

export const setCurrentPaletteId = (paletteId) => ({
  type: 'SET_CURRENT_PALETTE_ID', 
  paletteId
});

export const setCurrentProjectName = (projectName) => ({
  type: 'SET_CURRENT_PROJECT_NAME', 
  projectName
});

export const setRandomHexcodes = (hexcodes) => ({
  type: 'SET_RANDOM_HEXCODES', 
  hexcodes
});

export const toggleDisplayRandom = (shouldDisplay) => ({
  type: 'DISPLAY_RANDOM', 
  shouldDisplay
});
