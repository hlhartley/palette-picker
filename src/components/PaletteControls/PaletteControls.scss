.PaletteControls {
    border: 2px solid green;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.palette-header {
    color: grey;
}