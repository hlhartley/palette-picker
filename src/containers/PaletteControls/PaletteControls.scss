.PaletteControls {
    background-color: #e6f5f7;
    height: 7vh;
}

.palette-controls-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.palette-header {
    color: grey;
    font-family: 'Allerta Stencil', sans-serif;
    font-size: 25px;
    margin-top: 37px;
}

.generate-button {
    margin-top: 2px;
    margin-left: 6px;
    height: 36px;
    min-width: 88px;
    padding: 6px 16px;
    line-height: 1.5;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    touch-action: manipulation;
    cursor: pointer;
    border:0;
    border-radius: 2px;
    background: #e91e63;
    color:#fff;
    outline:0;
    &:hover {
        background: darken(#e91e63, 12%)  
    }
}

.md-btn-raised {
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 
                0 1px 5px 0 rgba(0, 0, 0, 0.12), 
                0 3px 1px -2px rgba(0, 0, 0, 0.2);
    transition: box-shadow 0.28s cubic-bezier(0.4, 0, 0.2, 1);

    &:active {
        box-shadow: 0 8px 10px 1px rgba(0, 0, 0, 0.14),
                0 3px 14px 2px rgba(0, 0, 0, 0.12),
                0 5px 5px -3px rgba(0, 0, 0, 0.4);
    }
}