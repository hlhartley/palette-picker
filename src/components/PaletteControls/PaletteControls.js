import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setRandomHexcodes, toggleDisplayRandom } from '../../actions/index';

export class PaletteControls extends Component {

  findPaletteName = () => {
    const  { palettes, currentPaletteId } = this.props;
    const matchingPalette = palettes.find((palette) => {
      return palette.id === currentPaletteId || { name: 'Palette' }
    })
    return matchingPalette.name;
  }

  generateRandomColors = () => {
    let possibleHexValues = "0123456789ABCDEF";
    let newColors = [];
    for (let i = 0; i < 5; i++) {
      let hexStringArray = [];
      for (let j = 0; j < 6; j++) {
        let generatedHexcode = possibleHexValues[(Math.floor(Math.random() * 16))]; 
        hexStringArray.push(generatedHexcode);
      }
      let hexcode = hexStringArray.join('');
      newColors.push(hexcode);
    }
    this.props.setRandomHexcodes(newColors);
    this.props.toggleDisplayRandom(true);
  }

  render() {
    return (
      <div className="PaletteControls">
        {this.props.palettes.length ? (
          <div className="palette-controls-container">
            <button className="generate-button" onClick={this.generateRandomColors}>Generate New Palette!</button>
            <h2 className="palette-header">{this.findPaletteName()}</h2>
          </div>
        ) : (
          <p>loading...</p>
        )}
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  palettes: state.palettes,
  currentPaletteId: state.currentPaletteId,
  randomHexcodes: state.randomHexcodes
});

export const mapDispatchToProps = (dispatch) => ({
  setRandomHexcodes: (hexcodes) => dispatch(setRandomHexcodes(hexcodes)),
  toggleDisplayRandom: (shouldDisplay) => dispatch(toggleDisplayRandom(shouldDisplay))
});

export default connect(mapStateToProps, mapDispatchToProps)(PaletteControls);
