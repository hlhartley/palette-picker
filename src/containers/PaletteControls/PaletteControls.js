import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setRandomHexcodes, toggleDisplayRandom } from '../../actions/index';
import LoadingDisplay from '../../components/LoadingDisplay/LoadingDisplay';

export class PaletteControls extends Component {

  componentDidMount = () => {
    this.generateRandomColors();
  }

  findPaletteName = () => {
    const  { palettes, currentPaletteId } = this.props;
    let matchingPalette = palettes.filter((palette) => {
      return palette.id === currentPaletteId
    })
    if (!matchingPalette.length) {
      matchingPalette = [{ name: 'Palette' }]
    }
    return matchingPalette[0].name.toUpperCase();
  }

  generateRandomColors = async () => {
    let possibleHexValues = "0123456789ABCDEF";
    let newColors = [];
    for (let i = 0; i < 5; i++) {
      let hexStringArray = [];
      for (let j = 0; j < 6; j++) {
        let generatedHexcode = possibleHexValues[(Math.floor(Math.random() * 16))]; 
        hexStringArray.push(generatedHexcode);
      }
      let hexcode = hexStringArray.join('');
      newColors.push(hexcode);
    }
    await this.props.setRandomHexcodes(newColors);
    await this.props.toggleDisplayRandom(true);
  }

  render() {
    if (this.props.palettes) {
      return (
        <div className="PaletteControls">
          {this.props.palettes.length ? (
            <div className="palette-controls-container">
              <button className="generate-button" onClick={this.generateRandomColors}>Generate New Palette!</button>
              <h2 className="palette-header">{this.findPaletteName()}</h2>
            </div>
          ) : (
            <LoadingDisplay/>
          )}
        </div>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  palettes: state.palettes,
  currentPaletteId: state.currentPaletteId,
  randomHexcodes: state.randomHexcodes
});

export const mapDispatchToProps = (dispatch) => ({
  setRandomHexcodes: (hexcodes) => dispatch(setRandomHexcodes(hexcodes)),
  toggleDisplayRandom: (shouldDisplay) => dispatch(toggleDisplayRandom(shouldDisplay))
});

export default connect(mapStateToProps, mapDispatchToProps)(PaletteControls);
