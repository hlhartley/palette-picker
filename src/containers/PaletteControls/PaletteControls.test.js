import React from 'react';
import { shallow } from 'enzyme';
import { PaletteControls, mapStateToProps, mapDispatchToProps } from './PaletteControls';
import { setHexcodes } from '../../actions';
import { mockPalettes } from '../../__fixtures__/mockData';

describe('PaletteControls', () => {
  let wrapper;
  let mockSetHexcodes;
  mockSetHexcodes = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <PaletteControls setHexcodes={mockSetHexcodes} palettes={mockPalettes} hexcodes={['ffffff', 'fffff1', 'fffff2', 'fffff3', 'fffff4', 'fffff5']}/>
      )
    });

    describe('PaletteControls component', () => {
      it('should properly render the component elements', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('mapStateToProps', () => {
      it('should return an object with palettes and hexcodes as keys', () => {
        const mockState = {
          palettes: [],
          extra: '',
          hexcodes: []
        }
        const expected = {
          palettes: [],
          hexcodes: []
        }
        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
      });
    });

    describe('mapDispatchToProps', () => {
      it('should call dispatch when setHexcodes is called', () => {
        const mockDispatch = jest.fn()
        const actionToDispatch = setHexcodes();
        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.setHexcodes()
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
      });
    });
});