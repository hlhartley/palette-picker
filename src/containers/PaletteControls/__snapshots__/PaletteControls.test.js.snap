// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteControls PaletteControls component should properly render the component elements 1`] = `""`;
